File: src/main/java/global/FileHandler.java
package global;

import java.awt.image.BufferedImage;
import java.io.File;
import java.io.IOException;
import javax.imageio.ImageIO;

public class FileHandler {

    public static File getFile() {
        while (true) {
            try {
                for (File file: (new File(G.USER_DIR)).listFiles()) {
                    if (isValidFile(file)) {
                        return file;
                    }
                }
            } catch (NullPointerException e) { }

            try {
                Thread.sleep(10_000);
            } catch (InterruptedException e) { }
        }
    }

    public static BufferedImage getImage(File file) {
        BufferedImage b = null;
        do {
            try {
                b =  ImageIO.read(file);
            } catch (IOException e) {
                System.out.println(G.FILE_ERROR + file.getName());
            }
        } while (b == null);
        return b;
    }

    public static void putImageInFile(File f, String folder, BufferedImage image, String append) {
        File fi = toFile(f, folder, append);
        try {
            if (!fi.exists()) {
                fi.mkdirs();
            }
            ImageIO.write(image, G.PNG, fi);
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }

    private static boolean isValidFile(File f) {
        if (f.isFile()) {

            final String ending = f.getName().substring(f.getName().lastIndexOf('.'));

            switch (ending) {
                case G.DOT_PNG:
                case G.DOT_JPG:
                case G.DOT_BMP:
                    return true;
            }
        }
        return false;
    }

    private static File toFile(File f, String folder, String append) {
        return new File(f.getParent() + File.separator +
                folder + File.separator +
                f.getName().substring(0, f.getName().length() - 4) + append + G.DOT_PNG);
    }
}

File: src/main/kotlin/global/FileHandler.kt
package global

import java.awt.image.BufferedImage
import java.io.File
import java.io.IOException
import javax.imageio.ImageIO

object FileHandler {

    fun getImageFile(): File? {
        return File(Global.USER_DIR).listFiles()?.find { file ->
            file.isValidFile()
        }
    }

    fun getImage(file: File): BufferedImage? {
        return try {
            ImageIO.read(file)
        } catch (e: IOException) {
            println(Global.FILE_ERROR + file.name)
            null
        }
    }

    fun putImageInFile(
        f: File,
        folder: String,
        image: BufferedImage?,
        append: String
    ) {
        val fi = toFile(f, folder, append)
        try {
            if (!fi.exists()) {
                fi.mkdirs()
            }
            ImageIO.write(image, Global.PNG, fi)
        } catch (e: IOException) {
            throw RuntimeException(e)
        }
    }

    private fun toFile(f: File, folder: String, append: String): File {
        return File(File(f.parent, folder), f.nameWithoutExtension + append + Global.DOT_PNG)
    }

    private fun File.isValidFile(): Boolean {
        return isFile && (
                extension.equals(Global.PNG, ignoreCase = true)
                        || extension.equals(Global.JPG, ignoreCase = true)
                        || extension.equals(Global.BMP, ignoreCase = true)
                )
    }
}